### DISCUSSION TITLE
Build Failed on centOS 7

### DISCUSSION THREAD

**Discussion by rithikN** [AcademySoftwareFoundation/OpenRV#66] [Build Issues]
If anyone please let me know how to fix below error. Please let me know if more info required.

[ 42%] Built target RvApp
[ 42%] Linking CXX executable ../../../../stage/app/bin/ApplicationTest
/opt/rh/devtoolset-9/root/usr/libexec/gcc/x86_64-redhat-linux/9/ld: /usr/lib/libGLX.so: error adding symbols: file in wrong format
collect2: error: ld returned 1 exit status
gmake[3]: *** [src/test/IPCore/ApplicationTest/CMakeFiles/ApplicationTest.dir/build.make:174: stage/app/bin/ApplicationTest] Error 1
gmake[2]: *** [CMakeFiles/Makefile2:14917: src/test/IPCore/ApplicationTest/CMakeFiles/ApplicationTest.dir/all] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:13794: src/bin/apps/rv/CMakeFiles/rv.dir/rule] Error 2
gmake: *** [Makefile:3820: rv] Error 2

**Comment by rogernelson** [comment]
Hi, it looks like it's try to link with a 32-bit graphics library instead of a 64-bit. The setup for this depends on which graphics card and/or drivers you have installed for it.  Do you have a 64-bit equivalent file (ie  /usr/lib64/libGLX.so)? If not, you probably need to install the drivers for whichever graphics card you are using. Or as an alternative, you could install the mesa software rendering library with `yum install mesa-libGL`, but this will have poorer performance compared to the proper drivers for your graphics card.
